scene/gui/tree.cpp

```diff
@@ -5245,6 +5243,13 @@ void Tree::_do_incr_search(const String &p_add) {
 	ensure_cursor_is_visible();
 }
 
+void Tree::_incr_search(const Ref<InputEventKey> &p_event_key) {
+	if (get_viewport()->is_input_handled()) {
+		return;
+	}
+	_do_incr_search(String::chr(p_event_key->get_unicode()));
```

Comment:
You could put it inside the condition. Early return for one line is not really worth it.